tests/lti/views/oauth_test.py

```diff
@@ -52,49 +53,33 @@ def test_it_gets_the_client_secret_from_AuthData(self, pyramid_request):
         code request.
 
         """
-        oauth.token_callback(pyramid_request)
+        method(pyramid_request)
 
         pyramid_request.auth_data.get_lti_secret.assert_called_once_with(
             'TEST_OAUTH_CONSUMER_KEY')
 
-    def test_it_gets_the_canvas_server_from_AuthData(self, pyramid_request):
+    def test_it_gets_the_canvas_server_from_AuthData(self, pyramid_request, method):
         """It gets the URL of the Canvas server to login to from AuthData."""
-        oauth.token_callback(pyramid_request)
+        method(pyramid_request)
 
         pyramid_request.auth_data.get_canvas_server.assert_called_once_with(
             'TEST_OAUTH_CONSUMER_KEY')
 
-    def test_it_posts_an_authorization_code_request_to_canvas(self,
-                                                              pyramid_request,
-                                                              requests_fixture):
-        oauth.token_callback(pyramid_request)
-
-        requests_fixture.post.assert_called_once_with(
-            'https://TEST_CANVAS_SERVER.com/login/oauth2/token', {
-                'code': 'TEST_OAUTH_AUTHORIZATION_CODE',
-                'grant_type': 'authorization_code',
-
-                # The client secret that we retrieved from AuthData.
-                'client_secret': 'TEST_CLIENT_SECRET',
-
-                'client_id': u'TEST_OAUTH_CONSUMER_KEY',
-                'redirect_uri': 'http://TEST_LTI_SERVER.com/token_init',
-        })
```

Comment:
This test has been moved to below, because it applies only to `token_callback()`